packages/devtools_app/lib/src/timeline/flutter/timeline_screen.dart

```diff
@@ -151,14 +130,71 @@ class TimelineScreenBodyState extends State<TimelineScreenBody>
         child: Row(
           children: [
             Switch(
-              value: controller.timelineMode == TimelineMode.frameBased,
+              value: mode == TimelineMode.frameBased,
               onChanged: _onTimelineModeChanged,
             ),
             const Text('Show frames'),
           ],
         ),
       ),
     ];
+    if (mode == TimelineMode.frameBased) {
+      return ValueListenableBuilder(
+        valueListenable: controller.frameBasedTimeline.pausedNotifier,
+        builder: (context, paused, _) {
+          return Row(
+            children: [
+              OutlineButton(
+                onPressed: paused ? null : _pauseLiveTimeline,
+                child: const MaterialIconLabel(
+                  Icons.pause,
+                  'Pause',
+                  minIncludeTextWidth: 900,
+                ),
+              ),
+              OutlineButton(
+                onPressed: !paused ? null : _resumeLiveTimeline,
+                child: const MaterialIconLabel(
+                  Icons.play_arrow,
+                  'Resume',
+                  minIncludeTextWidth: 900,
+                ),
+              ),
+              ...sharedWidgets,
+            ],
+          );
+        },
+      );
+    } else {
+      return ValueListenableBuilder(
+        valueListenable: controller.fullTimeline.recordingNotifier,
+        builder: (context, recording, _) {
+          return Row(
+            children: [
+              OutlineButton(
+                key: TimelineScreen.recordButtonKey,
+                onPressed: recording ? null : _startRecording,
+                child: const MaterialIconLabel(
+                  Icons.fiber_manual_record,
```

Comment:
This is a bit too complicated to read easily.  I suggest refactoring each if/else clause out into a helper method.

Reply:
cleaned up